### PR TITLE
D2B detector efficiency calculation

### LINKED ISSUE #21550: ILL Powder Diffraction, extend calibration for 2-D case
PowderDiffILLDetEffCorr algorithm needs to be extended to support the 2-dimensional case, e.g. D2B detector with position-sensitive-tubes.

### LINKED ISSUE #22093: Incorrect rounding in SumOverlappingTubes
### Expected behavior

Use `std::lround` and `boost::math::iround` to fix warnings from [misc-incorrect-roundings](http://releases.llvm.org/5.0.1/tools/clang/tools/extra/docs/clang-tidy/checks/misc-incorrect-roundings.html)

### Actual behavior

system test `PowderDiffILLDetScanReductionTest` fails when a naive rounding formula is replaced with `boost::math::iround`. (The naive formula fails when the integer is < 0)

### Steps to reproduce the behavior

1. Apply attached patch
2. Build mantid
3. run system test `PowderDiffILLDetScanReductionTest`

### Platforms affected

all

[SumOverlappingTubes.txt](https://github.com/mantidproject/mantid/files/1813141/SumOverlappingTubes.txt)

### PR DESCRIPTION
**Pull request by gvardany**
This PR extends the `PowderDiffILLDetEffCorr` algorithm to support the efficiency calculation for D2B instrument, which is a scanning powder diffractometer at the ILL. 

Some satellite algorithms are also fixed/adjusted correspondingly; there were minor bug fixes and extensions.

**To test:**
Use the files 532008 and 532009 from system test data under ILL/D2B directory (set facility to ILL) : 
```
PowderDiffILLDetEffCorr(CalibrationRun='532008,532009', DerivationMethod='GlobalSummedReference2D', OutputWorkspace='calib')
```
The output should be a 2D map of detector efficiencies, similar to the plot in the documentation.

Play around with the input properties of the algorithm, make sure the behaviour is reasonable. Also try to break it.

Make sure the documentation reads clear and the testing is adequate.

The method and the results are preliminarily agreed upon with the instrument scientist.

Fixes #21550 . 
Fixes #22093 .
Refs #21729 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [x] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] Do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/PowderDiffILLDetEffCorr-v1.rst [hunk 1]
```diff
@@ -9,84 +9,196 @@
...
+   from mantid.simpleapi import PowderDiffILLDetEffCorr
+   PowderDiffILLDetEffCorr(CalibrationRun='967076.nxs', DerivationMethod='SequentialSummedReference1D', OutputWorkspace='calib')
+   Transpose(InputWorkspace='calib', OutputWorkspace='calib')
+   fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
+   ax.plot(mtd['calib'],'-')
+   ax.set_xlabel('Pixel #')
+   ax.set_ylabel('Calibration constant')
+   fig.show()
+
+.. figure:: /images/D20_calib.png
```
Comment (soininen) [inline docs/source/algorithms/PowderDiffILLDetEffCorr-v1.rst]: Could the `.. plot:` directive be used here or does it take too long to produce the documentation?

File: docs/source/algorithms/PowderDiffILLDetEffCorr-v1.rst [hunk 2]
```diff
@@ -9,84 +9,196 @@
...
+
+For **D2B** it is assumed that the tubes and pixels pass through the exact same positions during the scan.
+That is, the tubes have to be aligned vertically and horizontally and the gap between each pair of neighboring tubes must be integer multiple of the scan step.
+
+D20 Workflow
+------------
+
+.. diagram:: PowderDiffILLDetEffCorr-v1_D20_wkflw.dot
+
+**Example - D20**
```
Comment (soininen) [inline docs/source/algorithms/PowderDiffILLDetEffCorr-v1.rst]: Is the example data available in Mantid's usage example data package? If so, it should be mentioned here as well. There is some sphinx directive to add a standard phrase and link here.

**Review by soininen — CHANGES_REQUESTED** [review]
Tested this using the instructions provided as well as with some data on ILL's network. `PowderDiffILLDetEffCorr` worked without problems and gave comprehensive error messages when suspicious input was given. There are a few comments I made on the code; additionally:
- *OutputWorkspace* could have X and Y axis labels if somebody wants to plot 2D.
- There is an extra dot in the doc string of the *PixelRange* property preventing the tooltip to show correctly.

**Comment by gvardany** [comment]
I think I have addressed all your comments now.

**Review by soininen — APPROVED** [review]

**Pull request merged**